### ISSUE TITLE
Linestyle kwarg ignored in lineplot

### ISSUE THREAD

**Issue by scratchmex** [mwaskom/seaborn#2094]
As seen in [this question](https://stackoverflow.com/questions/51963725/how-to-plot-a-dashed-line-on-seaborn-lineplot), the `linestyle` argument in `sns.lineplot` seems to be ignored in the final plot.
The code to reproduce
```python
import seaborn as sns

sns.lineplot(range(10), range(10), linestyle='--')
sns.lineplot(range(10), range(10), linestyle=':')
```
gives 
![gives](https://user-images.githubusercontent.com/4014888/82738871-985cc980-9d00-11ea-8187-aae5b24e454d.png)

The expected behavior should be
![expected](https://user-images.githubusercontent.com/4014888/82738857-79f6ce00-9d00-11ea-9c7c-7688f06d7622.png)
![expected](https://user-images.githubusercontent.com/4014888/82738817-40be5e00-9d00-11ea-919b-82f80028e726.png)

As said in the [documentation](https://seaborn.pydata.org/generated/seaborn.lineplot.html)

> Other keyword arguments are passed down to matplotlib.axes.Axes.plot().

Looking in the code, at https://github.com/mwaskom/seaborn/blob/2d3502b05d91343bd37d8f8074874cfbae9f1e68/seaborn/relational.py#L669 it seems that it does pass the keyword arguments.

**Comment by mwaskom** [comment]
Yes, this is the same issue as https://github.com/mwaskom/seaborn/issues/1526
